File: server/api/system/v1/social.go
// Package v1
// @Link  https://github.com/xiujiecn/xiu-admin
// @Copyright  Copyright (c) 2025 LiXiujie
// @Author  Lxj <li@xiujie.cn>
// @License  https://github.com/xiujiecn/xiu-admin/blob/master/LICENSE
package v1

import (
	"xiuadmin/internal/model"
	"xiuadmin/internal/model/request"
	"xiuadmin/internal/model/response"

	"github.com/gogf/gf/v2/frame/g"
)

type SysSocialListReq struct {
	g.Meta `path:"/social/list" method:"get" tags:"系统-第三方授权管理" summary:"社交列表" x-check-permission:"cpc:current:user"`
	model.SysSocialListParam
	request.PageInfo
}

type SysSocialListRes struct {
	Items []*model.SysSocialListModel `json:"items"`
	response.PageResult
}

// 删除绑定关系
type SysSocialDeleteReq struct {
	g.Meta `path:"/social/delete" method:"post" tags:"系统-第三方授权管理" summary:"删除社交" x-check-permission:"cpc:current:user"`
	Id     int64 `json:"id" dc:"主键"`
}

type SysSocialDeleteRes struct {
}

File: vue/apps/web-antd/src/api/system/social.ts
/**
 * @description 社交账号管理相关接口
 * @Link  https://github.com/xiujiecn/xiu-admin
 * @Copyright  Copyright (c) 2025 LiXiujie
 * @Author  Lxj <li@xiujie.cn>
 * @License  https://github.com/xiujiecn/xiu-admin/blob/master/LICENSE
 * @date 2024-03-21
 */

import { requestClient } from '#/api/request';

export interface SysSocialListReq {
  page?: number;
  pageSize?: number;
  userId?: number;
}

export interface SysSocialListRes {
  items: SysSocialListModel[];
  total: number;
}

export interface SysSocialListModel {
    id: number;
userId: number;
tenantId: string;
  authId: string;
  source: string;
  openId: string;
  userName: string;
  nickName: string;
  email: string;
  avatar: string;
  accessToken: string;
  expireIn: number;
  refreshToken: string;
  accessCode: string;
  unionId: string;
  scope: string;
  tokenType: string;
  idToken: string;
  macAlgorithm: string;
  macKey: string;
  code: string;
  oauthToken: string;
  oauthTokenSecret: string;
  createdDept: number;
  createdBy: number;
  createdAt: string;
}

export async function getSysSocialListApi(params: SysSocialListReq) {
    return requestClient.get<SysSocialListRes>('/system/social/list', { params });
}
